programs/stake/src/stake_state.rs

```diff
@@ -281,17 +316,17 @@ fn calculate_stake_rewards(
     vote_state: &VoteState,
     stake_history: Option<&StakeHistory>,
     inflation_point_calc_tracer: Option<impl Fn(&InflationPointCalculationEvent)>,
-    _fix_activating_credits_observed: bool, // this unused flag will soon be justified by an upcoming feature pr
+    credits_auto_rewind: bool,
 ) -> Option<(u64, u64, u64)> {
     // ensure to run to trigger (optional) inflation_point_calc_tracer
-    // this awkward flag variable will soon be justified by an upcoming feature pr
-    let mut forced_credits_update_with_skipped_reward = false;
-    let (points, credits_observed) = calculate_stake_points_and_credits(
-        stake,
-        vote_state,
-        stake_history,
-        inflation_point_calc_tracer.as_ref(),
-    );
+    let (points, credits_observed, mut forced_credits_update_with_skipped_reward) =
```

Comment:
a bit contrived flag thing. but i rather wanted some encapsulation for maintainability (place new code touched at lowest-level for all codepath even including currently-hypothetical code, which isn't via this `fn calculate_stake_rewards`) and separation-of-concerns (no visible muddy credits handling with feature activation here and put it into `calculate_stake_point_and_credits`)